services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -4957,6 +4950,27 @@ void preCheckStorePartitionCountUpdate(String clusterName, Store store, int newP
       throw new VeniceHttpException(HttpStatus.SC_BAD_REQUEST, errorMessage, ErrorType.INVALID_CONFIG);
     }
 
+    // Generally, a batch store should not have an RT topic, and an RT topic for a hybrid store should not exist.
+    // However, safeguards are necessary to prevent clients from toggling the hybrid setting with a partition update
+    // in edge cases.
+    if (!store.isHybrid()) {
+      PubSubTopic realTimeTopicForBatchStore = pubSubTopicRepository.getTopic(Utils.getRealTimeTopicName(store));
+      if (realTimeTopicForBatchStore != null) {
+        TopicManager topicManager = getTopicManagerForCluster(clusterConfig);
```

Comment:
Actually, I am thinking we can just fail if we see there is hybrid version exist. But since partition count is the same, maybe that's OK.
The only concern I can think of is they disable hybrid for tweaking partitioner implementation without restarting the producer, then the RT could be still mixed of record partitioned by old impl and new impl. 
But which might be a minor concern, because I think in that case, we anyway need to ask them to stop and restart the job, otherwise it will anyway be wrong. So anyway, it could be considered and handled separately